source/test/TestDirector.cc

```diff
@@ -4,21 +4,93 @@
 
 #include <abstract/File.hh>
 
+#include <egg/util/Stream.hh>
+
 #include <cstddef>
+#include <cstring>
 #include <format>
 
 namespace Test {
 
-TestDirector::TestDirector() {
+TestDirector::TestDirector(std::span<u8> &binaryData) : m_curTestIdx(0) {
+    Abstract::File::Remove("results.txt");
+    EGG::RamStream stream = EGG::RamStream(binaryData.data(), binaryData.size());
+    stream.setEndian(std::endian::big);
+    parseBinaryData(stream);
+    init();
+}
+
+TestDirector::~TestDirector() = default;
+
+void TestDirector::parseBinaryData(EGG::RamStream &stream) {
+    u16 numTestCases = stream.read_u16();
+    m_versionMajor = stream.read_u16();
+    m_versionMinor = stream.read_u16();
+
+    for (u16 i = 0; i < numTestCases; ++i) {
+        // Validate alignment
+        if (stream.read_u32() != 0x54535448) {
+            K_PANIC("Invalid binary data for test case!");
+        }
+
+        u16 totalSize = stream.read_u16();
+        TestCase testCase;
+
+        u16 nameLen = stream.read_u16();
+        std::string name = stream.read_string();
+        if (nameLen != name.size() + 1) {
+            K_PANIC("Test case name length mismatch!");
+        }
+
+        u16 rkgPathLen = stream.read_u16();
+        std::string rkgPath = stream.read_string();
+        if (rkgPathLen != rkgPath.size() + 1) {
+            K_PANIC("Test case RKG Path length mismatch!");
+        }
+
+        u16 krkgPathLen = stream.read_u16();
+        std::string krkgPath = stream.read_string();
+        if (krkgPathLen != krkgPath.size() + 1) {
+            K_PANIC("Test case KRKG Path length mismatch!");
+        }
+
+        char *nameArr = new char[nameLen];
+        strcpy(nameArr, name.c_str());
+        testCase.name = std::span(nameArr, nameLen);
+
+        char *rkgPathArr = new char[rkgPathLen];
+        strcpy(rkgPathArr, rkgPath.c_str());
+        testCase.rkgPath = std::span(rkgPathArr, rkgPathLen);
+
+        char *krkgPathArr = new char[krkgPathLen];
+        strcpy(krkgPathArr, krkgPath.c_str());
+        testCase.krkgPath = std::span(krkgPathArr, krkgPathLen);
```

Comment:
These are memory leaks with TestCase's default destructor.

Reply:
Just gonna make these `std::string` instead of char arrays.